include/deal.II/base/mpi.h

```diff
@@ -719,6 +719,22 @@ namespace Utilities
     MinMaxAvg
     min_max_avg(const double my_value, const MPI_Comm &mpi_communicator);
 
+    /**
+     * Same as above but returning the sum, average, minimum, maximum,
+     * process id of minimum and maximum as a collective operation of on the
+     * given MPI @ref GlossMPICommunicator "communicator" @p mpi_communicator
+     * for each entry of the vector.
+     *
+     * @note This function performs a single reduction sweep.
+     *
+     * @req Size of the input vector has to be the same on all processes.
+     */
+    void
+    min_max_avg(const ArrayView<const double> &my_values,
+                const ArrayView<MinMaxAvg> &   result,
```

Comment:
Oh, I had gotten that wrong because I had (wrongly) had the model of the `max` function in mind where the returned object is really just a single scalar of the same type as the input.

This works, but I would also have been ok if the function returned a `std::vector<MinMaxAvg>`. But this is fine by me.

Reply:
Given that we have been switching some other functions to return a vector in case we do some simple computation (like #9272), I would suggest we provide also this variant. I don't think it hurts to have yet another interface, as the implementation only redirects to this function.